File: CGStudio/PGCoreTest/Math/Circle3dTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PG.Core.Math;

namespace PGCoreTest.Math
{
    [TestClass]
    public class Circle3dTest
    {
        private readonly double tolerance = 1.0e-12;

        [TestMethod]
        public void TestGetPosition()
        {
            var circle = new Circle3d(new Vector3d(0, 0, 0), new Vector3d(1, 0, 0), new Vector3d(0, 1, 0));
            {
                var actual = circle.GetPosition(0.25);
                var expected = new Vector3d(0, 1, 0);
                Assert.IsTrue(expected.IsSame(actual, tolerance));
            }

            {
                var actual = circle.GetPosition(0.75);
                var expected = new Vector3d(0, -1, 0);
                Assert.IsTrue(expected.IsSame(actual, tolerance));
            }
        }

        [TestMethod]
        public void TestNormal()
        {
            var circle = new Circle3d(new Vector3d(0, 0, 0), new Vector3d(1, 0, 0), new Vector3d(0, 1, 0));
            var actual = circle.Normal();
            var expected = new Vector3d(0, 0, 1);
            Assert.IsTrue(expected.IsSame(actual, tolerance));
        }
    }
}

File: CGLib/Crystal/MathTest/Circle3dTest.cpp
#include "gtest/gtest.h"

#include "../Math/Circle3d.h"

using namespace Crystal::Math;

namespace {
	const double tolerance = 1.0e-12;
}

TEST(Circle3dTest, TestGetNormal)
{
	const Circle3d circle(Vector3dd(0, 0, 0), Vector3dd(1, 0, 0), Vector3dd(0, 1, 0));
	EXPECT_TRUE( areSame(Vector3dd(0,0,1), circle.getNormal(), tolerance) );
}

TEST(Circle3dTest, TestGetPosition)
{
	const Circle3d circle(Vector3dd(0, 0, 0), Vector3dd(1, 0, 0), Vector3dd(0, 1, 0));
	EXPECT_TRUE( areSame(Vector3dd(1,0,0), circle.getPosition(0.0), tolerance) );
	EXPECT_TRUE( areSame(Vector3dd(0,1,0), circle.getPosition(0.25), tolerance));
	EXPECT_TRUE( areSame(Vector3dd(-1,0,0), circle.getPosition(0.5), tolerance));
	EXPECT_TRUE(areSame(Vector3dd(0,-1, 0), circle.getPosition(0.75), tolerance));
	EXPECT_TRUE(areSame(Vector3dd(1, 0, 0), circle.getPosition(1.0), tolerance));

}